src/hooks/useProjectCreation.tsx

```diff
@@ -0,0 +1,200 @@
+import { useTranslation } from "react-i18next";
+import { useNavigate } from "react-router-dom";
+
+import { LoggerService } from "@services";
+import { defaultProjectFile, namespaces } from "@src/constants";
+import { ModalName } from "@src/enums/components";
+import { useFileOperations } from "@src/hooks";
+import { Manifest } from "@src/interfaces/models";
+import { FileStructure } from "@src/interfaces/utilities";
+import { unpackFileZip } from "@src/utilities";
+
+import { useModalStore, useProjectStore, useToastStore } from "@store";
+
+export const useProjectCreation = () => {
+	const { t } = useTranslation("dashboard", { keyPrefix: "templates" });
+	const { t: tUtil } = useTranslation("utilities", { keyPrefix: "fetchAndExtract" });
+	const { createProject, createProjectFromManifest, getProjectsList, pendingFile, projectsList, setPendingFile } =
+		useProjectStore();
+	const projectNamesSet = useMemo(() => new Set(projectsList.map((project) => project.name)), [projectsList]);
+	const navigate = useNavigate();
+	const addToast = useToastStore((state) => state.addToast);
+	const { closeModal, openModal } = useModalStore();
+
+	const [isCreatingNewProject, setIsCreatingNewProject] = useState(false);
+	const [loadingImportFile, setLoadingImportFile] = useState(false);
+	const [projectId, setProjectId] = useState<string>();
+	const { saveAllFiles } = useFileOperations(projectId || "");
+	const [templateFiles, setTemplateFiles] = useState<FileStructure>();
+	const fileInputRef = useRef<HTMLInputElement>(null);
+
+	const handleCreateProject = async (name: string) => {
+		setIsCreatingNewProject(true);
+		const { data, error } = await createProject(name, true);
+		setIsCreatingNewProject(false);
+
+		if (error) {
+			addToast({
+				message: t("errorCreatingProject"),
+				type: "error",
+			});
+
+			return { error: true };
+		}
+		const projectId = data?.projectId;
+		navigate(`/projects/${projectId}`, {
+			state: { fileToOpen: defaultProjectFile },
+		});
+
+		return { error: false };
+	};
+
+	useEffect(() => {
+		const getAndSaveFiles = async () => {
+			if (!projectId || !templateFiles) return;
+
+			const fileEntries = Object.entries(templateFiles).map(([path, fileNode]) => {
+				const content = "content" in fileNode ? fileNode.content : "";
+
+				return [path, new Uint8Array(new TextEncoder().encode(content))];
+			});
+
+			await saveAllFiles(Object.fromEntries(fileEntries));
+			addToast({
+				message: t("projectCreatedSuccessfully"),
+				type: "success",
+			});
+
+			getProjectsList();
+			navigate(`/projects/${projectId}`);
+		};
+
+		getAndSaveFiles();
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [projectId]);
+
+	const extractManifestFromFiles = async (
+		zipFile: File
+	): Promise<{ manifest: Manifest; structure: FileStructure } | null> => {
+		try {
+			const { error, structure } = await unpackFileZip(zipFile);
+			if (error) {
+				addToast({ message: tUtil("uknownErrorUnpackingZip"), type: "error" });
```

Comment:
Add a file name to toast and to the logger

Reply:
added